src/cowboy_compress_h.erl

```diff
@@ -72,21 +72,26 @@ early_error(StreamID, Reason, PartialReq, Resp, Opts) ->
 
 %% Check if the client supports decoding of gzip responses.
 check_req(Req) ->
-	%% @todo Probably shouldn't unconditionally crash on failure.
-	case cowboy_req:parse_header(<<"accept-encoding">>, Req) of
-		%% Client doesn't support any compression algorithm.
-		undefined ->
-			#state{compress=undefined};
-		Encodings ->
-			%% We only support gzip so look for it specifically.
-			%% @todo A recipient SHOULD consider "x-gzip" to be
-			%% equivalent to "gzip". (RFC7230 4.2.3)
-			case [E || E={<<"gzip">>, Q} <- Encodings, Q =/= 0] of
-				[] ->
-					#state{compress=undefined};
-				_ ->
-					#state{compress=gzip}
-			end
+	try
+		case cowboy_req:parse_header(<<"accept-encoding">>, Req) of
+			%% Client doesn't support any compression algorithm.
+			undefined ->
+				#state{compress=undefined};
+			Encodings ->
+				%% We only support gzip so look for it specifically.
```

Comment:
I thought about doing this one too, wdyt?

Reply:
No don't because that will change eventually.